ignite/metrics/precision.py

```diff
@@ -90,14 +90,23 @@ class Precision(_BasePrecisionRecall):
         In binary and multilabel cases, the elements of `y` and `y_pred` should have 0 or 1 values. Thresholding of
         predictions can be done as below:
 
-        .. code-block:: python
+        .. testcode::
 
-            def thresholded_output_transform(output):
+            def thresholded_output_transform(engine, output):
                 y_pred, y = output
                 y_pred = torch.round(y_pred)
                 return y_pred, y
+            engine = Engine(thresholded_output_transform)
+            metric = Precision()
```

Comment:
I'm not sure if this is what we want to expose to our users. `thresholded_output_transform` should remain the argument to `Precision` and not for `Engine`.

Reply:
Actually that should be `process_function`.

Reply:
The example shows how to binarize predictions for Precision metric only. Returning binarized preds from validator may be a limitation...

Reply:
In general we can improve the docstring for Precision (same for Accuracy and this comment: https://github.com/pytorch/ignite/pull/2287#discussion_r735169293) and show all possible cases: 

    binary case
    multiclass case
    multilabel case